plugins/repository-s3/build.gradle

```diff
@@ -169,177 +147,39 @@ buildscript {
-        script.setText(
-          ["function shutdown {",
-           "  kill ${minioPid}",
-           "}",
-           "trap shutdown EXIT",
-           // will wait indefinitely for input, but we never pass input, and the pipe is only closed when the build dies
-           "read line\n"].join('\n'), 'UTF-8')
-        final ProcessBuilder killer = new ProcessBuilder("bash", script.absolutePath)
-        killer.start()
-        break
-      }
-    }
-
-    if (httpPort <= 0) {
-      throw new IllegalStateException("httpPort must be > 0")
-    }
-  }
-}
-
-if (useFixture && minioDistribution) {
-  apply plugin: 'de.undercouch.download'
-
-  final String minioFileName = "minio.${minioVersion}"
-  final String minioDownloadURL = "https://dl.minio.io/server/minio/release/${minioDistribution}/archive/${minioFileName}"
-  final String minioFilePath = "${gradle.gradleUserHomeDir}/downloads/minio/${minioDistribution}/${minioFileName}"
+if (useFixture) {
 
-  task downloadMinio(type: Download) {
-    src minioDownloadURL
-    dest minioFilePath
-    onlyIfModified true
-  }
+  apply plugin: 'elasticsearch.test.fixtures'
 
-  task verifyMinioChecksum(type: Verify, dependsOn: downloadMinio) {
-    src minioFilePath
-    algorithm 'SHA-256'
-    checksum minioCheckSum
-  }
-
-  task installMinio(type: Sync, dependsOn: verifyMinioChecksum) {
-    from minioFilePath
-    into minioBinDir
-    fileMode 0755
+  RestIntegTestTask integTestMinio = project.tasks.create('integTestMinio', RestIntegTestTask.class) {
+    description = "Runs REST tests using the Minio repository."
   }
 
-  task startMinio {
-    dependsOn installMinio
-
-    ext.minioPid = 0L
-    ext.minioPort = 0
-
+  Task writeDockerFile = project.tasks.create('writeDockerFile') {
+    File minioDockerfile = new File("${project.buildDir}/minio-docker/Dockerfile")
+    outputs.file(minioDockerfile)
     doLast {
-      new File("${minioDataDir}/${s3PermanentBucket}").mkdirs()
-
-      Exception accumulatedException = null
-      for (int i = 0; i < 5; ++i) {
-        try {
-          minioPort = freePort(minioAddress)
-          final Process process =
-            setupMinio(minioAddress, minioPort, minioDataDir, s3PermanentAccessKey, s3PermanentSecretKey, minioBinDir, minioFileName)
-          minioPid = getMinioPid(process)
-          addShutdownHook(process, minioPort, minioPid)
-          break
-        } catch (Exception e) {
-          logger.error("Exception while trying to start Minio {}", e)
-          if (accumulatedException == null) {
-            accumulated = e
-          } else {
-            accumulatedException.addSuppressed(e)
-          }
-        }
-      }
-      if (accumulatedException != null) {
-        throw new GradleException("Failed to start Minio", accumulatedException)
-      }
+      minioDockerfile.parentFile.mkdirs()
+      minioDockerfile.text = "FROM minio/minio:RELEASE.2019-01-23T23-18-58Z\n" +
+        "RUN mkdir -p /minio/data/${s3PermanentBucket}\n" +
```

Comment:
I chose to dynamically build the `Dockerfile` here since we need to create this directory in the container somehow to force the existence of a bucket.

Reply:
Would it make sense to have hard coded values for bucket, access and secret keys in case we use minio so the test always runs ? Could also  be in a follow up, but that way we would not need a docker file at all  as we can just include the hard-coded values in the docker compose configuration and mount a volume that already has a directory for the bucket.  I don't think this is a blocker to merge this PR, but would be good to have this test running with the regular build.

Reply:
Do we want to use local fs volume mounts though? I'd rather not, that's gonna be a little annoying for environments (like my personal CI :D) that use Docker in Docker or a remote Docker daemon?

Reply:
I'm not sure this will work in docker based local CI right now and the tests will just be skipped.  I'm not sure we can avoid the volume mounts for all fixtures anyway, but maybe we can create the dirs as part of the start command ?